### PR TITLE
oboe-10.3.0

### PR DESCRIPTION
**Pull request by ronilan**
### Overview:
This pull request provides the bindings with the functionality required in order to work with liboboe 10.3.x.

Implementation is using minimal modification approach. No refactoring has been done.

### Main changes:
- init options include mode and options updated to version 13.
- Event::makeFromBuffer accepts either xtrace or traceparent formatted input and acts accordingly.
- getTraceSettings accepts either xtrace or traceparent formatted input and acts accordingly.
- getTraceSettings accepts a tracestate as input and passes to oboe tracing decision method. input updated to version 3.
- javascript Event.makeFromString (wrapper to Event::makeFromBuffer) accepts either xtrace or traceparent  formatted input and acts accordingly.
- Tightened regex validation of xtrace and traceparent.

### Notes: 
- This is a **non breaking change** (mode 0 is compatible with 10.2.0)
- Future changes will be merged to this branch.
- This branch is the base to main branch of: https://github.com/solarwindscloud/solarwinds-bindings-node
- This branch is being used by https://github.com/appoptics/appoptics-apm-node/tree/nh-main

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,24 +27,33 @@ try {
...
 // augment the c++ event object with a string conversion
 // function.
 //
+
+const validXtrace = (xtrace) => {
+  // https://github.com/librato/trace/tree/master/docs/specs
+  const regExp = /\b2B[0-9A-F]{40}[0-9A-F]{16}0[0-1]{1}\b/
+  const matches = regExp.exec(xtrace)
+
+  return matches ? matches[0] : ''
```
Comment (jerrytfleung) [inline index.js:43]: Question: Why here return matches[0] or '' but not the same return for `validTraceparent`?

File: src/settings.cc [hunk 2]
```diff
@@ -94,6 +94,7 @@ Napi::Value getTraceSettings(const Napi::CallbackInfo& info) {
   // in defaults
   bool have_metadata = false;
   std::string xtrace("");
```
Comment (jerrytfleung) [inline src/settings.cc:96]:
```suggestion
  std::string xtrace;
```

File: src/settings.cc [hunk 3]
```diff
@@ -94,6 +94,7 @@ Napi::Value getTraceSettings(const Napi::CallbackInfo& info) {
...
+  std::string tracestate("");
```
Comment (jerrytfleung) [inline src/settings.cc:97]:
```suggestion
  std::string tracestate;
```

File: index.js (hunk 1 above)
Comment (ronilan) [reply index.js:43]: Cleaned. Thanks for catching.

File: src/settings.cc (hunk 2 above)
Comment (ronilan) [reply src/settings.cc]: Fixed.

File: src/settings.cc (hunk 3 above)
Comment (ronilan) [reply src/settings.cc]: Fixed.

**Pull request closed without merging**